addons/calendar/data/mail_template_data.xml

```diff
@@ -75,7 +75,7 @@
             <ul>
                 <t t-if="object.event_id.location">
                     <li>Location: <t t-out="object.event_id.location or ''">Bruxelles</t>
-                        (<a target="_blank" t-attf-href="http://maps.google.com/maps?oi=map&amp;q={{object.event_id.location}}">View Map</a>)
+                        <a target="_blank" t-if="not is_online or is_online and object.event_id.location != object.event_id.appointment_type_id.location_id.name" t-attf-href="http://maps.google.com/maps?oi=map&amp;q={{object.event_id.location}}">(View Map)</a>
```

Comment:
Why comparing event location (a char field) to appointment type location (a res.partner address or name) ? Are we really going to have something that is similar ?

Reply:
I realise the condition is a bit confusing at the first glance :smile: 

The following code of `appointment_type.py` computes the field location_id for appointment. Now in case of no contact address set (i.e. our targetted use case here), it sets the name as location.  
```
@api.depends('location_id')
def _compute_location(self):
  """Use location_id if available, otherwise its name, finally ''. """
  for record in self:
     if (record.location_id.contact_address or '').strip():
             record.location = ', '.join(
                 frag.strip()
                 for frag in record.location_id.contact_address.split('\n') if frag.strip()
             )
      else:
             record.location = record.location_id.name or ''
```
Hence, here I compare that if the event location is not equal to JUST the name (has a contact address as well apart from a location name i.e. the IF part of above code would have been executed) , only in that case, the map hyperlink should be displayed. :sweat_smile:

Reply:
I understood it, but what are the odds that the location on event (char field) are exactly matching location_id.name ?

Reply:
Acutally I discussed it with JEH too. There are some edge cases here so I want to clear up some points first before tweaking the code :sweat_smile: 

The spec says we need to display the hyperlink only if location has an address set, not just a name. 

But we always display the `calendar.event.location` in all templates. Hence:

**Case 1:** 
In case of [non-appointment type](https://www.veed.io/view/39165f25-e557-402d-9ab9-7ecb58fde2cc?panel=quality-survey) events (base calendar), even if location has just a name or any random letters, the link will be displayed because there's nothing to identify a proper address since it's a simple char field entered manually by user.

**Case 2:**
For events of [type appointment](https://www.veed.io/view/bdbcc29b-4ce4-4a30-b007-c3d3bef68c42?panel=share), `location` of calender.event has a correct address (the one computed as `location_id` for appointment). But then even in cases when a proper address has been set, if user alters the `location` field, then the altered value (say 'abc') will be sent as appointment location instead of the set address.

Also as you said, as per my current condition, even if address isn't set and user alters calendar location, the strings would not match and the link would be displayed.

Now for base calendar (non-appointment),  JEH has suggested we keep the behavior as it is and always display the link no matter the value of location field. But in case of appointment, the location is computed properly. Now, since this task is now targeting master, can't we then make it a readonly so that user cannot change `location` field directly and it's computed from `location_id`?

Let me know your insights on how we want to proceed with and without appointment...